### DISCUSSION TITLE
Hide application menus when showing menu bar items

### DISCUSSION THREAD

**Discussion by bode-p** [jordanbaird/Ice#110] [General]
Can someone help me understand the option 'Hide application menus when showing menu bar items'?

Thanks in advance!

**Comment by jordanbaird** [comment]
If you have a lot of menu bar items and they overlap with the application menus (File, Edit, etc.), macOS doesn't show the items, instead giving the application menus priority. If you enable the `Hide application menus when showing menu bar items` feature, Ice will attempt to hide the application menus when the items in the hidden or always-hidden sections are shown. It does this by activating itself without showing its window. This shows Ice's own application menu, which is empty, freeing up more room for the shown items.

- Note: I say "attempt" because macOS Sonoma doesn't always allow an app to activate when it requests it. Not really a bug in Ice, per se, but a quirk in the way macOS behaves.
